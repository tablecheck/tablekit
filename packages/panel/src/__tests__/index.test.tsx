import { mount } from '@tablecheck/tablekit-enzyme';

import { Panel } from '../index';

beforeEach(() => {
  global.console.error = jest.fn();
});

describe('Panel', () => {
  test('should be possible to create a component', () => {
    const wrapper = mount(<Panel isOpen togglePanel={() => {}} />);
    expect(wrapper).not.toBeUndefined();
  });

  test('should correctly apply styling', () => {
    expect(mount(<Panel isOpen togglePanel={() => {}} />)).toMatchSnapshot();
  });

  test('should allow passing of custom className', () => {
    expect(
      mount(<Panel isOpen togglePanel={() => {}} className="test-class" />)
    ).toMatchSnapshot();
  });
});
