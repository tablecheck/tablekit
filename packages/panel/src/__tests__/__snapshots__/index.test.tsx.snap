// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should allow passing of custom className 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed!important;
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.77, 0.2, 0.05, 1);
  transition: transform 0.3s cubic-bezier(0.77, 0.2, 0.05, 1);
  right: 0;
  top: 0;
  -webkit-transform: translateX(0);
  -moz-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  width: 100%;
  height: 100%;
  box-shadow: 0px 4px 12px 0px rgba(0, 0, 0, 0.1);
  background-color: white;
  z-index: 300;
  box-shadow: 0 6px 16px 0 rgba(0, 0, 0, 0.1);
}

<ForwardRef
  className="test-class"
  isOpen={true}
  togglePanel={[Function]}
>
  <StaticPanelContainer
    className="test-class"
    data-ignore="scroll-lock-ignore"
    height="100%"
    isOpen={true}
    onClick={[Function]}
    position="right"
    width="100%"
  >
    <div
      className="test-class emotion-0 emotion-1"
      data-ignore="scroll-lock-ignore"
      onClick={[Function]}
    />
  </StaticPanelContainer>
</ForwardRef>
`;

exports[`Panel should correctly apply styling 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed!important;
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.77, 0.2, 0.05, 1);
  transition: transform 0.3s cubic-bezier(0.77, 0.2, 0.05, 1);
  right: 0;
  top: 0;
  -webkit-transform: translateX(0);
  -moz-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  width: 100%;
  height: 100%;
  box-shadow: 0px 4px 12px 0px rgba(0, 0, 0, 0.1);
  background-color: white;
  z-index: 300;
  box-shadow: 0 6px 16px 0 rgba(0, 0, 0, 0.1);
}

<ForwardRef
  isOpen={true}
  togglePanel={[Function]}
>
  <StaticPanelContainer
    data-ignore="scroll-lock-ignore"
    height="100%"
    isOpen={true}
    onClick={[Function]}
    position="right"
    width="100%"
  >
    <div
      className="emotion-0 emotion-1"
      data-ignore="scroll-lock-ignore"
      onClick={[Function]}
    />
  </StaticPanelContainer>
</ForwardRef>
`;
